docs/ballooning.md

```diff
@@ -0,0 +1,257 @@
+# Using the balloon device with Firecracker
+
+## What is the balloon device
+
+A memory balloon device is a virtio device that can be used to reclaim and
+give back guest memory through API commands issued by the host. It does this
+by allocating memory in the guest, and then sending the
+addresses of that memory to the host; the host may then remove that memory at
+will. The device is configured through a number of options, and an integer,
+which represents the target size of the balloon, in MiB. The options cannot be
+changed during operation, but the target size can be changed.
+
+The behaviour of the balloon is the following: while the actual size of the
+balloon (i.e. the memory it has allocated) is smaller than the target
+size, it continually tries to allocate new memory -- if it fails, it prints an
+error message (`Out of puff! Can't get %d pages`), sleeps for 0.2 seconds, and
+then tries again. While the actual
+size of the balloon is larger than the target size, it will free memory until
+it hits the target size. The following options are available:
```

Comment:
Are the options related to how balloon is freeing memory? Otherwise please put the following options... in a new paragraph. Also rephrase to the balloon device can be configured with the following options

There is a line break nit above while the actual....